src/bundle/upload.ts

```diff
@@ -291,3 +299,78 @@ export const uploadDeprecatedCommand = async (apikey: string, options: Options)
     program.error('')
   }
 }
+
+// eslint-disable-next-line max-len
+async function checkNativeCode(channel: string, appId: string, bundle: string, supabase: SupabaseClient<Database>): Promise<{nativeFilesNotChanged: boolean, hashes: string[]}> {
+  const nativeFilesSpinner = p.spinner()
+  nativeFilesSpinner.start('Calculating checksum of native files')
+  const nativeCodeLocal = await findAllNativeCode()
+  nativeFilesSpinner.stop(`Native code length: ${nativeCodeLocal.length}`);
+
+  const hashes = nativeCodeLocal.map(val => val.hash)
+
+  const {data: dataChannel, error: errorChannel } = await supabase
+    .from('channels')
+    .select(`
+      version (
```

Comment:
you should get the last version uploaded not the channel

Reply:
Well... this raises an interesting question. How do we handle manual updates? Let's say user wants to downgrade from `2.0.0` -> `1.0.5`. With this code he could do that, even tho the native files might be different and the whole downgrade would require new app store version. But if he does complete this downgrade and we get the latest version (not the channel as you suggested) the we would be getting the `2.0.0` version even tho the app would be on `1.0.5`.

I believe getting this from the channel is not a terrible solution